contracts/paymaster/BasePaymaster.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: GPL-3.0-only
+// COPIED FROM: https://github.com/opengsn/gsn/blob/master/packages/contracts/src/BasePaymaster.sol
+// Changes:
+// - Drops Ownable (OpenZeppelin).
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+
+import "@opengsn/contracts/src/interfaces/IPaymaster.sol";
+import "@opengsn/contracts/src/interfaces/IRelayHub.sol";
+import "@opengsn/contracts/src/forwarder/IForwarder.sol";
+import "@opengsn/contracts/src/utils/GsnEip712Library.sol";
+
+/**
+ * @notice An abstract base class to be inherited by a concrete Paymaster.
+ * A subclass must implement:
+ *  - preRelayedCall
+ *  - postRelayedCall
+ */
+abstract contract BasePaymaster is IPaymaster, ERC165 {
+  using ERC165Checker for address;
+
+  IRelayHub internal relayHub;
+  address private _trustedForwarder;
+
+  /// @inheritdoc IPaymaster
+  function getRelayHub() public view override returns (address) {
+    return address(relayHub);
+  }
+
+  //overhead of forwarder verify+signature, plus hub overhead.
+  uint256 public constant FORWARDER_HUB_OVERHEAD = 50000;
```

Comment:
Are those the real costs? I thought it was much cheaper to go with this pattern. Oh well.

Reply:
Everything in `BasePaymaster.sol` is copied from their version: https://github.com/opengsn/gsn/blob/master/packages/contracts/src/BasePaymaster.sol. Why is it copied? Their version uses `Ownable` from OZ which I didn't want to bring in.

They're (OpenGSN's) implementation instructions say to grab `BasePaymaster.sol` and base our base paymaster from that. See here for sample paymaster contracts: https://github.com/opengsn/gsn/tree/master/packages/paymasters/contracts